File: adv_2022/adv_day_02/cpp/day_02.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <map>

int main(int argc, char **argv){
    std::ifstream File("input.txt");
    std::string moves;

    // Move Points: Rock = 1 Point, Paper = 2 Point, Scissors = 3 Point
    std::map<std::string, int> guideline{{"Rock", 1}, {"Paper", 2}, {"Scissors", 3}};
    // Result Points: Win = 6 Point, Draw = 3 Point, Lose = 0 Point
    std::map<std::string, int> result_points{{"Win", 6}, {"Draw", 3}, {"Lose", 0}};
    // A Rock, B Paper, C Scissors
    // X Rock, Y Paper, Z Scissors
    std::map<std::string, std::string> encryption{{"A", "Rock"}, {"X", "Rock"},
                                                  {"B", "Paper"}, {"Y", "Paper"},
                                                  {"C", "Scissors"}, {"Z", "Scissors"}};
    int elf_point = 0;
    int our_point = 0;
    int total_result_point = 0;

    while (std::getline(File, moves)) {
        std::string move[2];
        int i = 0;
        std::stringstream ssin(moves);
        while (ssin.good() && i < 2){
            ssin >> move[i];
            ++i;
        }

        elf_point = guideline[encryption[move[0]]];
        our_point = guideline[encryption[move[1]]];

        if (elf_point == our_point) {
            total_result_point += our_point + result_points["Draw"];
        }
        else if (( elf_point + our_point ) % 2 == 0) {
            if (elf_point < our_point) {
                total_result_point += our_point + result_points["Lose"];
            }
            else {
                total_result_point += our_point + result_points["Win"];
            }
        }
        else {
            if(our_point > elf_point) {
                total_result_point += our_point + result_points["Win"];
            }
            else {
                total_result_point += our_point + result_points["Lose"];
            }
        }

    }

    std::cout << total_result_point << std::endl;
    // 11386
    File.close();
}

File: adv_2022/adv_day_02/python/day_02.py

with open("input.txt") as f:
    moves = f.readlines()
# Move Points: Rock = 1 Point, Paper = 2 Point, Scissors = 3 Point
guideline = {"Rock": 1, "Paper": 2, "Scissors": 3}
# A Rock, B Paper, C Scissors
# X Rock, Y Paper, Z Scissors
encryption = {"A": "Rock", "X": "Rock",
              "B": "Paper", "Y": "Paper",
              "C": "Scissors", "Z": "Scissors"}
# Result Points: Win = 6 Point, Draw = 3 Point, Lose = 0 Point
result_points = {"Win": 6, "Draw": 3, "Lose": 0} 

total_result_point = 0

for move in moves:
    strategy_guide = move.split(" ")
    strategy_guide[1] = strategy_guide[1][0]

    elf_point = guideline[encryption[strategy_guide[0]]]
    our_point = guideline[encryption[strategy_guide[1]]]

    if(elf_point == our_point):
        total_result_point += our_point + result_points["Draw"]
    elif(( elf_point + our_point ) % 2 == 0):
        if(elf_point < our_point):
            total_result_point += our_point + result_points["Lose"]
        else:
            total_result_point += our_point + result_points["Win"]
    else:
        if(our_point > elf_point):
            total_result_point += our_point + result_points["Win"]
        else:
            total_result_point += our_point + result_points["Lose"]

print(total_result_point)
# 11386